app/services/analytics_service/src/services/background_tasks.py

```
@@ -0,0 +1,475 @@
+            period_start = target_datetime.replace(minute=0, second=0, microsecond=0)
+            period_end = period_start + timedelta(hours=1)
+        elif period == 'day':
+            period_start = target_datetime.replace(hour=0, minute=0, second=0, microsecond=0)
+            period_end = period_start + timedelta(days=1)
+        elif period == 'week':
+            days_since_monday = target_datetime.weekday()
+            period_start = target_datetime.replace(hour=0, minute=0, second=0, microsecond=0) - timedelta(days=days_since_monday)
+            period_end = period_start + timedelta(weeks=1)
+        else:  # month
+            period_start = target_datetime.replace(day=1, hour=0, minute=0, second=0, microsecond=0)
+            next_month = period_start.replace(month=period_start.month + 1) if period_start.month < 12 else period_start.replace(year=period_start.year + 1, month=1)
```

Pylint (3.3.7)

[convention] Line too long (165/100)

(C0301)